packages/core/crates/core-crypto/src/utils.rs

```diff
@@ -0,0 +1,16 @@
+/// Convenience method to XOR one slice onto other.
+pub fn xor_inplace(a: &mut [u8], b: &[u8]) {
+    let bound = a.len().min(b.len());
```

Comment:
Probably worth to `assert!(a.len() == b.len()`. Within the JS implementation, this has revealed various errors